### PR TITLE
testutil: have File* checker produce more useful error output

### PR DESCRIPTION
**Pull request by bboozzoo**
When working with File{Contains,Equals,Matches} checkers, the error output is not
very useful, eg:
```
backend_test.go:199:
    c.Check(fn, testutil.FileEquals, expected)
... filename string = "/tmp/check-5309021883990244937/1/var/lib/snapd/mount/snap.snap-name_instance.fstab"
... contents string = ""

backend_test.go:203:
    c.Check(fn, testutil.FileEquals, expected)
... filename string = "/tmp/check-5309021883990244937/1/var/lib/snapd/mount/snap.snap-name_instance.user-fstab"
... contents string = ""
```

The checker will now return the actual file contents that did not match, thus
the error output looks like this now:

```
backend_test.go:203:
    c.Check(fn, testutil.FileEquals, expected)
... filename string = "/tmp/check-3565332118040638512/1/var/lib/snapd/mount/snap.snap-name_instance.user-fstab"
... contents string = ""
... Failed to match with file content:
$USER_HOME_SNAP_DIR/snap-name_instance $USER_HOME_SNAP_DIR/snap-name none rbind,x-snapd.origin=parallel-instance 0 0
/src-3 /dst-3 none bind,ro 0 0
```

It would be nicer to include output formatting code from gopkg.in/check.v1 but
that is package private.

### CODE REVIEW

File: testutil/checkers_test.go [hunk 1]
```diff
@@ -36,8 +36,7 @@ import (
 	"testing"
 
 	"gopkg.in/check.v1"
-
-	"github.com/snapcore/snapd/strutil"
+	// "github.com/snapcore/snapd/strutil"
```
Comment (mvo5) [inline testutil/checkers_test.go]: Why is this commented out? If we don't need it we can just remove it I guess?

**Comment by zyga** [comment]
It seems we need to vendor more things:
```
src/github.com/snapcore/snapd/vendor/gopkg.in/check.v1/checkers.go:9:2: cannot find package "github.com/kr/pretty" in any of:
	/home/gopath/src/github.com/snapcore/snapd/_build/src/github.com/snapcore/snapd/vendor/github.com/kr/pretty (vendor tree)
```

File: testutil/checkers_test.go (hunk 1 above)
Comment (bboozzoo) [reply testutil/checkers_test.go]: Uhh, I'll force push to drop this one.

File: packaging/fedora/snapd.spec [hunk 2]
```diff
@@ -150,6 +150,8 @@ BuildRequires: golang(github.com/godbus/dbus)
 BuildRequires: golang(github.com/godbus/dbus/introspect)
 BuildRequires: golang(github.com/gorilla/mux)
 BuildRequires: golang(github.com/jessevdk/go-flags)
+BuildRequires: golang(github.com/kr/pretty)
```
Comment (zyga) [inline packaging/fedora/snapd.spec]: I'm happy to see those are available in the archive!

File: testutil/checkers.go [hunk 3]
```diff
@@ -78,28 +78,41 @@ func fileContentCheck(filename string, content interface{}, exact bool) (result
 	if err != nil {
 		return false, fmt.Sprintf("Can't read file %q: %v", filename, err)
 	}
+	presentableBuf := string(buf)
 	if exact {
 		switch content := content.(type) {
 		case string:
-			return string(buf) == content, ""
+			result = string(buf) == content
```
Comment (zyga) [inline testutil/checkers.go]: You have `string(buf)` above as `presentableBuf` (nitpick)

File: testutil/checkers.go [hunk 4]
```diff
@@ -78,28 +78,41 @@ func fileContentCheck(filename string, content interface{}, exact bool) (result
...
 		case []byte:
-			return bytes.Equal(buf, content), ""
+			result = bytes.Equal(buf, content)
+			presentableBuf = "<binary data>"
 		case fmt.Stringer:
-			return string(buf) == content.String(), ""
+			result = string(buf) == content.String()
```
Comment (zyga) [inline testutil/checkers.go]: Same here

File: testutil/checkers.go [hunk 5]
```diff
@@ -78,28 +78,41 @@ func fileContentCheck(filename string, content interface{}, exact bool) (result
...
+		default:
+			error = fmt.Sprintf("Can't compare file contents with something of type %T", content)
```
Comment (zyga) [inline testutil/checkers.go]: `cannot compare ...`

File: testutil/checkers.go [hunk 6]
```diff
@@ -78,28 +78,41 @@ func fileContentCheck(filename string, content interface{}, exact bool) (result
...
+			result = bytes.Contains(buf, content)
+			presentableBuf = "<binary data>"
 		case *regexp.Regexp:
-			return content.Match(buf), ""
+			result = content.Match(buf)
 		case fmt.Stringer:
-			return strings.Contains(string(buf), content.String()), ""
+			result = strings.Contains(string(buf), content.String())
+		default:
+			error = fmt.Sprintf("Can't compare file contents with something of type %T", content)
```
Comment (zyga) [inline testutil/checkers.go]: `cannot compare ...`

**Pull request merged**